seq/alphabet_conv.go

```diff
@@ -0,0 +1,638 @@
+}
+
+//https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=tgencodes#SG30
+var rnaCodonToAminoAcidTranslTable30 = map[string]string{
+	"UUU": "F", "UCU": "S", "UAU": "Y", "UGU": "C",
+	"UUC": "F", "UCC": "S", "UAC": "Y", "UGC": "C",
+	"UUA": "L", "UCA": "S", "UAA": "G", "UGA": "*",
+	"UUG": "L", "UCG": "S", "UAG": "G", "UGG": "W",
+	"CUU": "L", "CCU": "P", "CAU": "H", "CGU": "R",
+	"CUC": "L", "CCC": "P", "CAC": "H", "CGC": "R",
+	"CUA": "L", "CCA": "P", "CAA": "Q", "CGA": "R",
+	"CUG": "L", "CCG": "P", "CAG": "Q", "CGG": "R",
+	"AUU": "I", "ACU": "T", "AAU": "N", "AGU": "S",
+	"AUC": "I", "ACC": "T", "AAC": "N", "AGC": "S",
+	"AUA": "I", "ACA": "T", "AAA": "K", "AGA": "R",
+	"AUG": "M", "ACG": "T", "AAG": "K", "AGG": "R",
+	"GUU": "V", "GCU": "A", "GAU": "D", "GGU": "G",
+	"GUC": "V", "GCC": "A", "GAC": "D", "GGC": "G",
+	"GUA": "V", "GCA": "A", "GAA": "E", "GGA": "G",
+	"GUG": "V", "GCG": "A", "GAG": "E", "GGG": "G",
+}
+
+//https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=tgencodes#SG31
+var rnaCodonToAminoAcidTranslTable31 = map[string]string{
+	"UUU": "F", "UCU": "S", "UAU": "Y", "UGU": "C",
+	"UUC": "F", "UCC": "S", "UAC": "Y", "UGC": "C",
+	"UUA": "L", "UCA": "S", "UAA": "G", "UGA": "Y",
+	"UUG": "L", "UCG": "S", "UAG": "G", "UGG": "W",
+	"CUU": "L", "CCU": "P", "CAU": "H", "CGU": "R",
+	"CUC": "L", "CCC": "P", "CAC": "H", "CGC": "R",
+	"CUA": "L", "CCA": "P", "CAA": "Q", "CGA": "R",
+	"CUG": "L", "CCG": "P", "CAG": "Q", "CGG": "R",
+	"AUU": "I", "ACU": "T", "AAU": "N", "AGU": "S",
+	"AUC": "I", "ACC": "T", "AAC": "N", "AGC": "S",
+	"AUA": "I", "ACA": "T", "AAA": "K", "AGA": "R",
+	"AUG": "M", "ACG": "T", "AAG": "K", "AGG": "R",
+	"GUU": "V", "GCU": "A", "GAU": "D", "GGU": "G",
+	"GUC": "V", "GCC": "A", "GAC": "D", "GGC": "G",
+	"GUA": "V", "GCA": "A", "GAA": "E", "GGA": "G",
+	"GUG": "V", "GCG": "A", "GAG": "E", "GGG": "G",
+}
+
+// RNAToProtein accepts a RNA Sequence, translation table from NIBH standard and returns a protein sequence
+func RNAToProtein(rna []byte, transl_table int) (protein []byte, err error) {
+
+	if rna == nil {
+		return nil, errors.New("seq.RNAToProtein: input sequence is nil")
+	}
+	var acceptable_transl_tables []int
+	acceptable_transl_tables = append(acceptable_transl_tables, 1, 2, 3, 4, 5, 6, 9, 10, 11, 12, 13, 14, 16, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31)
+	var found bool = false
+	for _, t := range acceptable_transl_tables {
+		if transl_table == t {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil, errors.New("seq.RNAToProtein: translation table is not found")
+	}
+
+	var last3NTs bytes.Buffer
+	for _, nt := range rna {
+
+		err := last3NTs.WriteByte(nt)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(last3NTs.String()) == 3 {
+			switch transl_table {
+			case 1:
+				if aminoacid, ok := rnaCodonToAminoAcidTranslTable1[last3NTs.String()]; ok {
+					protein = append(protein, []byte(aminoacid)...)
```

Comment:
- again, appending to a `[]byte` without initialize capacity is ineffective. `bytes.Buffer` is an alternative.
- `[]byte(aminoacid)` costs some time if the value of condon table is `string`, `byte` is better.